diff -abBur source.orig/CompoMaster/CompoMaster.java source/CompoMaster/CompoMaster.java
--- source.orig/CompoMaster/CompoMaster.java	Mon Jan  9 22:01:17 2006
+++ source/CompoMaster/CompoMaster.java	Tue Jan 10 23:29:46 2006
@@ -31,7 +31,7 @@
             UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
         }
         catch(Exception exception) { }
-        URL url = (CompoMaster.CompoMaster.class).getResource("/cm_icon.gif");
+        URL url = (CompoMaster.class).getResource("/cm_icon.gif");
         System.out.println("".concat(String.valueOf(String.valueOf(url))));
         icon = Toolkit.getDefaultToolkit().getImage(url);
         cmpFilter = new CmpFilter();
diff -abBur source.orig/CompoMaster/SignupAdmin.java source/CompoMaster/SignupAdmin.java
--- source.orig/CompoMaster/SignupAdmin.java	Mon Jan  9 22:11:03 2006
+++ source/CompoMaster/SignupAdmin.java	Wed Jan 11 00:11:02 2006
@@ -186,7 +186,7 @@
             if(col == 1)
             {
                 int thisRank = data.getPlayer(row).getRank();
-                if(thisRank == 257)
+                if(thisRank == Data.MAXRANK)
                     return "N/A";
                 else
                     return new Integer(data.getPlayer(row).getRank());
@@ -1324,9 +1324,9 @@
             txtRank.setText("N/A");
         if(rank < 0)
             txtRank.setText("256");
-        if(rank == 257)
+        if(rank == Data.MAXRANK)
             txtRank.setText("N/A");
-        if(rank > 257)
+        if(rank > Data.MAXRANK)
             txtRank.setText("1");
     }
 
@@ -1550,6 +1550,7 @@
         boolean ok = false;
         File f = new File(txtChnFilename.getText());
         impData = Data.load(String.valueOf(CompoMaster.loadedFileDir) + String.valueOf(f.getName()));
+        JOptionPane.showMessageDialog(this, String.valueOf(CompoMaster.loadedFileDir) + String.valueOf(f.getName()), "Wrong file type", 0);
         mslQualify.removeAreas();
         if(impData != null)
         {
diff -abBur source.orig/Data/Data.java source/Data/Data.java
--- source.orig/Data/Data.java	Mon Jan  9 22:15:52 2006
+++ source/Data/Data.java	Tue Jan 10 23:51:56 2006
@@ -284,7 +284,7 @@
         }
 
         player = new ArrayList(numPlayers);
-        for(int rk = 1; rk <= 257; rk++)
+        for(int rk = 1; rk <= MAXRANK; rk++)
         {
             for(int i = 0; i < numPlayers; i++)
                 if(tempPlayer[i].getRank() == rk)
@@ -413,7 +413,7 @@
                 filename = String.valueOf(String.valueOf(filename)).concat(".cmp");
             FileOutputStream destination = new FileOutputStream(filename);
             ObjectOutputStream output = new ObjectOutputStream(destination);
-            output.writeObject("2.9.99");
+            output.writeObject(VERSION_NO);
             output.writeObject(d);
             output.flush();
             destination.close();
@@ -583,8 +583,8 @@
     public static final int COUNTMAPS = 0;
     public static final int COUNTFRAGS = 1;
     public static final boolean drawSignature = true;
-    public static final String VERSION_NO = "2.9.99";
-    public static final String VERSION = new String("Compo Master 2.9.99, Frode Nilsen, 2003");
+    public static final String VERSION_NO = "2.9.99.01";
+    public static final String VERSION = new String("Compo Master " + VERSION_NO + ", Frode Nilsen, 2003 + VVD patches, 2006");
     public static final String VERSION_2 = new String("http://www.compomaster.com.");
     public static final boolean ANONYMOUS = true;
     public static final int VERSIONTEXT_HEIGHT = 15;
diff -abBur source.orig/Data/DataInterface.java source/Data/DataInterface.java
--- source.orig/Data/DataInterface.java	Mon Jan  9 22:15:57 2006
+++ source/Data/DataInterface.java	Tue Jan 10 22:56:04 2006
@@ -463,7 +463,7 @@
 
     protected static void addPlayer()
     {
-        int rank = 257;
+        int rank = Data.MAXRANK;
         if(value.length > 2)
             try
             {
diff -abBur source.orig/Data/LoserOf.java source/Data/LoserOf.java
--- source.orig/Data/LoserOf.java	Mon Jan  9 22:16:45 2006
+++ source/Data/LoserOf.java	Tue Jan 10 22:56:10 2006
@@ -28,7 +28,7 @@
 
     public int getRank()
     {
-        return 257;
+        return Data.MAXRANK;
     }
 
     private Player getParentLoser()
diff -abBur source.orig/Data/MatchList.java source/Data/MatchList.java
--- source.orig/Data/MatchList.java	Mon Jan  9 22:16:53 2006
+++ source/Data/MatchList.java	Tue Jan 10 22:53:24 2006
@@ -165,7 +165,7 @@
         {
             g.setColor(new Color(128, 128, 128));
             MatchList _tmp = this;
-            String line = String.valueOf(String.valueOf((new StringBuffer(String.valueOf(String.valueOf(Data.VERSION_2)))).append(" v").append("2.9.99")));
+            String line = String.valueOf(String.valueOf((new StringBuffer(String.valueOf(String.valueOf(Data.VERSION_2)))).append(" v").append(Data.VERSION_NO)));
             g.drawString(line, di.width - g.getFontMetrics(getMainFont()).stringWidth(line) - 5, di.height - 6);
         }
     }
diff -abBur source.orig/Data/Player.java source/Data/Player.java
--- source.orig/Data/Player.java	Mon Jan  9 22:17:09 2006
+++ source/Data/Player.java	Tue Jan 10 22:56:18 2006
@@ -17,7 +17,7 @@
 
     public Player()
     {
-        rank = 257;
+        rank = Data.MAXRANK;
     }
 
     public Match getParentMatch()
diff -abBur source.orig/Data/RealPlayer.java source/Data/RealPlayer.java
--- source.orig/Data/RealPlayer.java	Mon Jan  9 22:17:22 2006
+++ source/Data/RealPlayer.java	Tue Jan 10 23:03:48 2006
@@ -30,7 +30,7 @@
         disabled = false;
         this.password = password;
         this.name = name;
-        super.rank = 257;
+		super.rank = Data.MAXRANK;
     }
 
     public RealPlayer(String name)
@@ -39,7 +39,7 @@
         disabled = false;
         this.name = name;
         password = null;
-        super.rank = 257;
+		super.rank = Data.MAXRANK;
     }
 
     public boolean isDisabled()
@@ -111,6 +111,7 @@
         try
         {
             Team t = null;
+			RealPlayer p = null;
             do
             {
                 String s;
@@ -169,9 +170,22 @@
                     String s;
                     if((s = inFile.readLine()) == null)
                         break;
-                    if(s.length() > 0)
-                        data.addPlayer(new RealPlayer(s));
+					if(s.startsWith("[playerdef]"))
+					{
+						if(p != null)
+							data.addPlayer(p);
+						p = new RealPlayer("");
+					}
+					String values[];
+					if((values = StringTools.eqSplit("name", '&', s)) != null)
+						p.setName(values[0]);
+					if((values = StringTools.eqSplit("password", '&', s)) != null)
+						p.setPassword(values[0]);
+					if((values = StringTools.eqSplit("rank", '&', s)) != null)
+						p.setRank(Integer.parseInt(values[0]));
                 } while(true);
+				if(p != null)
+					data.addPlayer(p);
             }
             Data data1 = data;
             return data1;
@@ -218,9 +232,14 @@
             } else
             {
                 outFile.write("[playerlist]\r\n");
-                Team team = data.getNumPlayers();
+				int team = data.getNumPlayers();
                 for(int i = 0; i < team; i++)
+				{
+					outFile.write("[playerdef]\r\nname=");
                     outFile.write(String.valueOf(String.valueOf(((RealPlayer)data.getPlayer(i)).name)).concat("\r\n"));
+					outFile.write("rank=");
+					outFile.write(String.valueOf(String.valueOf(((RealPlayer)data.getPlayer(i)).rank)).concat("\r\n"));
+				}
 
             }
             outFile.flush();
diff -abBur source.orig/Data/Walkover.java source/Data/Walkover.java
--- source.orig/Data/Walkover.java	Mon Jan  9 22:17:27 2006
+++ source/Data/Walkover.java	Tue Jan 10 22:56:46 2006
@@ -18,7 +18,7 @@
 
     public Walkover()
     {
-        super.rank = 257;
+        super.rank = Data.MAXRANK;
     }
 
     public Walkover(int rank)
